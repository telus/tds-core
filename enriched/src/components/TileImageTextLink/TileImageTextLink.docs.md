# TileImageTextLink Component
Component with stacked image, title, description, and link.

## Default (Left-aligned image)
Example:
```javascript
import TileImageTextLink from './TileImageTextLink';

// Example of data structure that needs to be supplied to Component
const data = {
  url: 'https://www.telus.com/mobility/phones/google-pixel', // required, text string
  title: 'Google Pixel', // optional, text string
  description: 'Manage your smart home products on Pixel and power them with the Google Home.', // required, text string
  image: { // Standard media field from Contentful
    file: {
      url: 'https://images.contentful.com/oia8yg65m0on/1Jxq0Bxmy0QGwgKQCG4UmU/9814d03e4cb8cc811e9b7bf4dc79ce71/google-pixel_2x.jpg'
    },
    description: 'Haut-parleurs internes de Google Home'
  }
};

const shopText = 'Shop Now';

<TileImageTextLink content={data} buttonText={shopText} />
```

```
<TileImageTextLink content={{
  url: 'https://www.telus.com/mobility/phones/google-pixel', // required, text string
  title: 'Google Pixel', // optional, text string
  description: 'Manage your smart home products on Pixel and power them with the Google Home.', // required, text string
  image: { // Standard media field from Contentful
    file: {
      url: 'https://images.contentful.com/oia8yg65m0on/1Jxq0Bxmy0QGwgKQCG4UmU/9814d03e4cb8cc811e9b7bf4dc79ce71/google-pixel_2x.jpg'
    },
    description: 'Haut-parleurs internes de Google Home'
  }
}} buttonText="Shop Now" />
```
