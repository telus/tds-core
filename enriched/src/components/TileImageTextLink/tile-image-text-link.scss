@import '../../../styles/core';

.tile-image-text-link {
  padding: 0 1rem;

  &:not(:last-child) {
    margin-bottom: 2rem;
  }

  @include from-breakpoint(small) {
    position: relative;
    order: 0;
    flex: 0 1 auto;
    align-self: stretch;
    padding-bottom: 1.25rem;

    &:not(:last-child) {
      margin-bottom: 0;
    }
  }
}

.tile-image-text-link__title {
  margin-top: 1rem;

  @include from-breakpoint(small) {
    margin-top: 1.5rem;
  }
}

.tile-image-text-link__link {
  @include from-breakpoint(small) {
    position: absolute;
    left: 1rem;
    bottom: 0;
    margin: 0 !important;
  }
}

.tile-image-text-link__image {
  width: 100%;
}
