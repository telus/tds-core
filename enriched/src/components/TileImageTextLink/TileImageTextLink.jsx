// Dependencies
import React from 'react';
import PropTypes from 'prop-types';

// Assets
import './tile-image-text-link.scss';

const TileImageTextLink = ({ content, buttonText, className }) => {
  return (
    <div className={`tile-image-text-link ${className}`}>
      <a tabIndex={-1} href={content.url}>
        <img
          className="tile-image-text-link__image"
          src={content.image.file.url} alt={content.image.description} />
      </a>
      {
        content.title &&
        <h4 className="tile-image-text-link__title" tabIndex="0">{content.title}</h4>
      }
      <p className="text--medium">{content.description}</p>
      <a
        href={content.url}
        className="text--medium chevron-link--primary tile-image-text-link__link"
        role="button">
        {buttonText}
      </a>
    </div>
  );
};

TileImageTextLink.propTypes = {
  content: PropTypes.object.isRequired,
  buttonText: PropTypes.string.isRequired,
  className: PropTypes.string
};


export default TileImageTextLink;
