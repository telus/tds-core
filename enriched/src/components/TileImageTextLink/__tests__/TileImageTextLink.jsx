import React from 'react';
import { shallow, render } from 'enzyme';
import { renderToJson } from 'enzyme-to-json';
import TileImageTextLink from '../TileImageTextLink';

describe('TileImageTextLink', () => {
  const content = {
    url: 'url',
    image: {
      file: {
        url: 'file url'
      },
      description: 'file description'
    },
    description: 'description'
  };
  const buttonText = 'Shop Now';

  const tileImageTextLink = shallow(
    <TileImageTextLink content={content} buttonText={buttonText} />);

  it('should render ImageText5050 (snapshot)', () => {
    expect(renderToJson(tileImageTextLink)).toMatchSnapshot();
  });

  it('should return a title if it\'s in the data', () => {
    content.title = 'title';

    const newTileImageTextLink = shallow(
      <TileImageTextLink content={content} buttonText={buttonText} />);

    expect(newTileImageTextLink.find('.tile-image-text-link__title').length).toEqual(1);
  });
});
