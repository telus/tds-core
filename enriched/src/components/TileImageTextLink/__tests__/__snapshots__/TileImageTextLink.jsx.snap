// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TileImageTextLink should render ImageText5050 (snapshot) 1`] = `null`;
